File: 3683-find-the-lexicographically-largest-string-from-the-box-i/3683-find-the-lexicographically-largest-string-from-the-box-i.cpp
class Solution {
public:
    string answerString(string word, int numFriends) {
        if (numFriends == 1) return word;
        int n = word.size();
        int k = n-numFriends+1;
        string ans = "";
        
        for(int i=0; i<word.size(); i++){
            ans = max(ans, word.substr(i, k));
        }
        
        return ans;
    }
};

File: 3683-find-the-lexicographically-largest-string-from-the-box-i/3683-find-the-lexicographically-largest-string-from-the-box-i.rs
impl Solution {
    pub fn answer_string(word: String, num_friends: i32) -> String {
        if num_friends == 1 {
            return word;
        }
        let n = word.len();
        let mut cur = String::new();
        let mut m = String::new();
        let split_length = n - num_friends as usize;
        for i in 0..n {
            let end = std::cmp::min(n, i + split_length + 1);
            cur = word[i..end].to_string();
            if cur > m {
                m = cur;
            }
        }
        m
    }
}